crates/maybenot/src/framework.rs

```diff
@@ -205,6 +208,30 @@ where
         self.current_time = current_time;
         for e in events {
             self.process_event(e);
+
+            // handle internal signaling
+            // Special case: if a machine sends a signal which results in another
+            // signal in response, that response may or may not be received; this
+            // depends on the order in which machines are given to the framework.
+            // Check for this explicitly (keep track of excluded machines).
+            if let Some(signal) = self.signal_pending {
+                let mut excluded = None;
+
+                for mi in 0..self.runtime.len() {
+                    if signal.is_some() && signal.unwrap() == mi {
+                        excluded = Some(mi);
+                        continue;
+                    }
+                    self.transition(mi, Event::Signal);
+                }
+
+                if let Some(excluded) = excluded {
+                    if self.signal_pending.unwrap().is_none() {
```

Comment:
You already have access to the signal in the variable `signal` here. So instead of this, would not just `signal.is_none()` work? If you do use my `Option::take` recommendation in the other comment, this would not work any longer, because the `unwrap` would panic since the value is already taken out of the pending signal.

Reply:
I made the same comment to Ethan at first that we already have access to the signal, but it turns out that the signal is copied there (I think?), and when `signal_pending` is set again in transition, we need to check for it again.

Regardless, I refactored this part, and it looks clearer now. Please let me know if you agree.

Reply:
That was indeed the issue - without accessing signal_pending directly, we didn't get the updated signal. The refactor also solves that problem, though, and I think it makes things look cleaner.